### PR TITLE
feat: prototype periodic check if gdc case-id-cache is stale and re-c…

### LINKED ISSUE #1872: gdc case id caching: periodically check api and recache if outdated

### PR DESCRIPTION
**Pull request by xzhou82**
…ache

## Description
closes #1872 
1. changed got to ky
2. auto-check every 5 min. only check total number of cases. awaiting method to check total number of aliquots and cases with exp data
3. on everyone's system can add `noPeriodicCheckGdcCaseCache` feature flag to disable checking

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [ ] Tests: added and/or passed unit and integration tests, or N/A
- [ ] Todos: commented or documented, or N/A
- [ ] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: server/src/termdb.gdc.js [hunk 1]
```diff
@@ -146,6 +145,8 @@ re.expand: []
 
 // prefix for keys in re._mapping{}
 const mapping_prefix = 'ssm_occurrence_centrics'
+// wait time for next check on stale case-id cache, 5min
+const cacheCheckWait = 5 * 60 * 1000
```
Comment (siosonel) [inline server/src/termdb.gdc.js:149]: @xzhou82 should make it configurable with default, `const cacheCheckWait = serverconfig.features.gdcCacheCheckWait || 5 * 60 * 1000`

**Comment by siosonel** [comment]
@xzhou82 I wonder if we can also use the response from `https://api.gdc.cancer.gov/status` to determine the need to update the cache.

Screenshot 2024-07-22 at 10 34 13 AM

**Comment by xzhou82** [comment]
inquired Phil about it and wait

**Comment by xzhou82** [comment]
ok to merge now. once the /status api is changed, will update method. don't want to hold the pr for it

**Pull request merged**